packages/graphql-hooks/src/Middleware.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Generic Middleware function that
+ * will run object through all provided functions
+ *
+ *
+ * Minimal example:
+ * const MyMiddleware = ({ operation }, next) => {
+ *   operation.variables.user = 'admin'
+ *   next()
+ * }
+ *
+ * All parameters provided are written in run function JSDoc
+ */
+export default class Middleware {
+  constructor(fns) {
+    if (fns.length === 0) {
+      // Pass through
+      fns.push((_, next) => next())
+    }
+
+    for (const fn of fns) {
+      if (typeof fn !== 'function') {
+        throw new Error(
+          'GraphQLClient Middleware: middleware has to be of type `function`'
+        )
+      }
+
+      this.run = (stack => (opts, next) => {
```

Comment:
Would be nice to avoid anonymous functions in order to debug (maybe even test it tbh) easier.
```
      at apply (packages/graphql-hooks/test/unit/Middleware.test.js:6:11)
      at apply (packages/graphql-hooks/src/Middleware.js:31:16)
      at stack (packages/graphql-hooks/src/Middleware.js:57:10)
      at stack (packages/graphql-hooks/src/Middleware.js:30:11)
      at Middleware.run (packages/graphql-hooks/src/Middleware.js:30:11)
      at Object.<anonymous> (packages/graphql-hooks/test/unit/Middleware.test.js:35:7)
```

Reply:
Good idea. Will create a new issue for this as the main priority is now to support APQ middleware